documentation/api/admin/v1/clean.markdown

```diff
@@ -0,0 +1,58 @@
+---
+title: "PuppetDB 4.1: Clean endpoint"
+layout: default
+canonical: "/puppetdb/latest/api/admin/v1/clean.html"
+---
+
+[curl]: ../curl.html#using-curl-from-localhost-non-sslhttp
+
+The `/clean` endpoint can be used to trigger PuppetDB maintenance
+operations.  Only one maintenance operation can be running at a time.
+Any request received while an operation is already in progress will
+return an HTTP conflict status (409).
+
+## `POST /pdb/admin/v1/clean`
+
+The maintanance operations must be triggered by a POST.
+
+### Request format
+
+The POST request should specify `Content-Type: application/json` and
+the request body should look like this:
+
+  ```json
+  {"version" : 1, "payload" : [REQUESTED_OPERATION ...]}
+  ```
+
+where valid `REQUESTED_OPERATION`s are `"expire-nodes"`, `"purge-nodes"`,
+`"purge-reports"`, and `"other"`.  An empty payload vector requests
+all maintenance operations.
```

Comment:
This likely requires discussion - presumably it should be underscores, but then should the corresponding PuppetDBService clean method take dashes or underscores, and/or keywords/strings?